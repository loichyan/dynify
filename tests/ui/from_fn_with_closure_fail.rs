fn main() {
    let var = 123;
    dynify::from_fn!(move || var);
}
