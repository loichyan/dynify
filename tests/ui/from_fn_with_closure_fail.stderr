error[E0308]: arguments to this function are incorrect
 --> tests/ui/from_fn_with_closure_fail.rs:3:5
  |
3 |     dynify::from_fn!(move || var);
  |     ^^^^^^^^^^^^^^^^^-------^^^^^
  |                      |
  |                      the expected closure
  |                      the found closure
  |
note: expected fn pointer, found closure
 --> tests/ui/from_fn_with_closure_fail.rs:3:5
  |
3 |     dynify::from_fn!(move || var);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: expected fn pointer `fn(dynify::function::MustNotBeClosure) -> {closure@$DIR/tests/ui/from_fn_with_closure_fail.rs:3:22: 3:29}`
                found closure `{closure@$DIR/src/function.rs:187:17: 187:18}`
note: closures can only be coerced to `fn` types if they do not capture any variables
 --> tests/ui/from_fn_with_closure_fail.rs:3:30
  |
3 |     dynify::from_fn!(move || var);
  |                              ^^^ `var` captured here
note: expected fn pointer, found closure
 --> tests/ui/from_fn_with_closure_fail.rs:3:5
  |
3 |     dynify::from_fn!(move || var);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: expected fn pointer `for<'a> unsafe fn(Slot<'a>, ()) -> &'a mut Opaque<_>`
                found closure `{closure@$DIR/src/function.rs:189:17: 189:37}`
note: closures can only be coerced to `fn` types if they do not capture any variables
 --> tests/ui/from_fn_with_closure_fail.rs:3:30
  |
3 |     dynify::from_fn!(move || var);
  |                              ^^^ `var` captured here
note: function defined here
 --> src/function.rs
  |
  | pub unsafe fn from_bare_fn<F, Args, Ret>(
  |               ^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::__from_fn` which comes from the expansion of the macro `dynify::from_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
